### ISSUE TITLE
Watching changes in the library throws an error when the application is served (Consuming your library during development).

### ISSUE THREAD

**Issue by raphael-volt** [jvandemo/generator-angular2-library#136]
Context :
* Consuming the library in development mode `npm link`
* Changes in the dist directory of the library are watched `npm run build:watch`
* An application that consumes the library is served `ng serve`

Make some changes in a file of the src directory of the library.

As the clean:dist task deletes all files in the dist directory, the index.js file is missing,so webpack can't load it to compile :
```bash
ERROR in .-lib/dist/index.js
Module build failed: Error: ENOENT: no such file or directory, open 'LIBRARY-DIRECTORY/dist/index.js'
 @ ./src/app/app.module.ts 10:0-44
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
 webpack: Failed to compile.
 ```

 Fast fix : editing the library gulpfile:

 ```javascript
/**
 * 1. Delete /dist folder
 */
gulp.task('clean:dist', function () {

  // Delete contents but not dist folder to avoid broken npm links
  // when dist directory is removed while npm link references it.
  return deleteFolders([distFolder + '/**', '!' + distFolder, '!' + distFolder + "/index.js"]);
});
 ```

 Watch changes an serve the application again.

 Make another changes in a file of the src directory of the library.

 webpack will successfully compile.

**Comment by alokstar** [comment]
@raphael-volt : Yes, same happened with me! That's why I have created the separate issue for this as 'testing as a standalone application' and waiting on the same. I know its pain while development but still a workaround is restart the application server or if you are still seeing the errors, just publish your package and then consume it as a dependency and it should work fine. For now, I am following the same approach for development. 
Hope it helps!

**Comment by jvandemo** [comment]
@raphael-volt &mdash; Thank you for sharing your workaround.

Unfortunately, we can't implement this behaviour by default, because we don't want any *old*  files lingering in the `dist` directory. If the build would fail, the `dist` directory would still contain the `index.js` file generated by the previous run.

Let's leave this issue open for a couple of weeks to see if someone else comes up with a possible solution. Thanks again! 👍

**Comment by UnEpicFail** [comment]
@raphael-volt Thanks for this solution, it was very useful.

**Comment by Janatbek** [comment]
I am having same issue, my consuming app fails in compilation. Any workarounds for assets folder?
We should write script to look the changes only. And leave whole dist folder as is. While developing in build:watch task not in build task

**Comment by jvandemo** [comment]
We hope to merge the playground soon. See #146 👍 

I will update this thread once the playground is available.

**Comment by jvandemo** [comment]
You can now preview library code changes live in the browser. The playground has been merged and released as v11.3.0 👍
